_applications/bad-apples.adoc

```diff
@@ -0,0 +1,310 @@
+  you can run the filter service, you will need to load the OpenShift
+  manifest that contains its template. Loading this manifest will also cause
+  OpenShift to begin building the service.
++
+....
+oc create -f https://raw.githubusercontent.com/radanalyticsio/bad-apples/master/filter/filter.yaml
+....
++
+To inspect the build for the watcher you can type `oc logs -f bc/watcher`,
+alternatively if you would like to check on the build status you can type
+`oc get build filter-1`.
+
+[[usage]]
+== Usage
+
+To utilize the Bad Apples project means to run the filter job and observe
+the results. As stated in the link:#architecture[Architecture] section,
+the filter job will read the transaction records from the database and then
+process them to determine which are fraudulent. It does this every time it
+is run. To aid in observing how many records the filter job has selected you
+will use the watcher service to inspect the database updates.
+
+. Vist the watcher with a browser to determine initial state. Open a web
+  browser and navigate to the route exposed by OpenShift for the watcher. The
+  URL address for the watcher can be found with the following command:
++
+....
+oc get route watcher
+....
++
+Given that the filter job has not run yet, you should see zero transactions
+detected as fraudulent. The page should look similar to this:
++
+pass:[<img src="/assets/howdoi/bad-apples-watcher1.png" alt="Initial watcher page" class="img-responsive">]
+
+. Run the filter job to process the transactions. You will now use the template
+  created during the installation with the following command:
++
+....
+oc new-app --template=filter \
+  -p IMAGE=$(oc get is filter --template={{.status.dockerImageRepository}}) \
+  -p OSHINKO_CLUSTER_NAME=sparky
+....
++
+The two parameters that are passed to this command (`IMAGE` and
+`OSHINKO_CLUSTER_NAME`) deserve a closer inspection. The `IMAGE` parameter
+instructs OpenShift where to find the built image for our filter job. As we
+used OpenShift to build the image, we are using the `oc get is filter` command
+to help find the reference for that image. This is done to accomodate for
+differing versions of OpenShift, specifically because while
+https://docs.okd.io/latest/dev_guide/managing_images.html#using-is-with-k8s[OpenShift supports using ImageStreams with Kubernetes objects]
+this support may not be in the version you have deployed. The
+`OSHINKO_CLUSTER_NAME` parameter gives us the opportunity to name the Spark
+cluster that will be automatically spawned for our job and thus we can reuse
+this cluster name with subsequent runs of the job.
++
+As the job is running you may wish to monitor its progress. To do this you
+can follow the logs that are produced by the job. You will need to use the
+job name that is generated by OpenShift and given in the output from the
+previous command. Once you have the job name, you will type
+`oc logs -f job <job name>` where `<job name>` is replaced by the actual name.
```

Comment:
Running on et0, at no point can I get this command to return anything other than as shown in the attached image 
<img width="366" alt="screen shot 2018-10-19 at 11 31 58" src="https://user-images.githubusercontent.com/5311289/47228320-ed04c980-d392-11e8-800b-3405cda01300.png">

Probably an error on my part...

Reply:
hmm, looks like i screwed up the command, should be `oc logs -f jobs/jdsl-av4g`

good spot, i'll fix it